unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -6936,129 +6936,149 @@ public bool objectIsCurrentlyVisible(SimObjPhysics sop, float maxDistance)
 
+                    foreach (float rotation in rotations) {
+                        transform.rotation = Quaternion.Euler(new Vector3(0.0f, rotation, 0.0f));
+
+                        foreach (Vector3 position in filteredPositions) {
+                            transform.position = position;
                             if (objectIsCurrentlyVisible(theObject, maxVisibleDistance)) {
-                                goodLocationsDict["x"].Add(p.x);
-                                goodLocationsDict["y"].Add(p.y);
-                                goodLocationsDict["z"].Add(p.z);
-                                goodLocationsDict["rotation"].Add(90.0f * i);
-                                goodLocationsDict["standing"].Add((1 - j) * 1.0f);
-                                goodLocationsDict["horizon"].Add(m_Camera.transform.localEulerAngles.x);
+                                validAgentPoses["x"].Add(position.x);
+                                validAgentPoses["y"].Add(position.y);
+                                validAgentPoses["z"].Add(position.z);
+                                validAgentPoses["rotation"].Add(rotation);
+                                validAgentPoses["standing"].Add(standing);
+                                validAgentPoses["horizon"].Add(horizon);
 
-#if UNITY_EDITOR
-                                // In the editor, draw lines indicating from where the object was visible.
-                                Debug.DrawLine(p, p + transform.forward * (gridSize * 0.5f), Color.red, 20f);
-#endif
+                                #if UNITY_EDITOR
+                                    // In the editor, draw lines indicating from where the object was visible.
+                                    Debug.DrawLine(position, position + transform.forward * (gridSize * 0.5f), Color.red, 20f);
+                                #endif
                             }
                         }
                     }
                 }
             }
 
-            if (wasStanding) {
-                stand();
-            } else {
-                crouch();
-            }
+            // restore old agent pose
+            if (wasStanding) stand(); else crouch();
             transform.position = oldPosition;
             transform.rotation = oldRotation;
             m_Camera.transform.localEulerAngles = oldHorizon;
             if (ItemInHand != null) {
                 ItemInHand.gameObject.SetActive(true);
             }
 
-#if UNITY_EDITOR
-            Debug.Log(goodLocationsDict["x"].Count);
-            Debug.Log(goodLocationsDict["x"]);
-#endif
+            #if UNITY_EDITOR
+                Debug.Log(validAgentPoses["x"].Count);
+                Debug.Log(validAgentPoses["x"]);
+            #endif
+
+            actionFinished(true, validAgentPoses);
+        }
+
+        [ObsoleteAttribute(message: "This action is deprecated. Call GetInteractablePositions instead.", error: false)]
```

Comment:
This is super useful with a nice IDE (i.e., vscode). It basically gives you an inline warning that the method you're using is deprecated.